### ISSUE TITLE
New rule: disallow ignored display values

### ISSUE THREAD

**Issue by MattDiMu** [stylelint/stylelint#2054]
There are some cases, where certain display values are overridden by other properties:

1. Elements with `position: absolute` or `position: fixed` (I'm not sure about `position: sticky`) have automatically applied `display: block` on them (if display: none is not present). Therefore CSS like this makes no sense:

`.selector {
  position: absolute;
  display: inline-block;
}`
If you open your developer tools, you will see, that the "computed" `display`-value of such an element is `block`.

=> new rule: `no-ignored-display-values`, which shows by default a warning, if a different value other than `block` or `none` is used within the same selector. There might even be an option, which disallows the use of `block` as well, as it's applied anyway and is therefore unnecessary. Only display: none is useful (currently hidden elements, which are shown later).

2. Elements with a float value other than `none` (which is either `left`, `right`, `inline-start` or `inline-end`) have usually a different display property applied as well. This is more complicated though. While most elements have their display value changed to `block`, it may also be changed to `table` (if the display value used to be `inline-table`), or isn't changed at all (if the display value was `flex` or `flex-inline`). A full "transformation"-table may be found here: https://developer.mozilla.org/en-US/docs/Web/CSS/float

I've experienced, that such errors are surprisingly common. Even though there are edge cases, where such combinations might make sense (e.g. if a different class overrides the position property to static again under certain circumstances, so that the `display: inline-block` does something), I feel that these occasions are very very rare and this constellation should therefore show a warning.

Edit: I just found a similar rule, which does such checks. As far as i see, it doesn't check for the two mentioned cases yet: https://github.com/stylelint/stylelint/tree/master/src/rules/declaration-block-no-ignored-properties

**Comment by evilebottnawi** [comment]
@MattDiMu format you issue as template, issue (example new rule) should contain valid and invalid `css` and why

**Comment by MattDiMu** [comment]
Sorry, I thought it would be much clearer this way, as the template is quite bug-oriented and it's a new feature, not a bug. Therefore all the questions about "reproducing" make no sense, while some information is missing. I've tried to "template-ify" the issue. 

> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New feature / extending rule "declaration-block-no-ignored-properties".

> Which rule, if any, is this issue related to?

https://github.com/stylelint/stylelint/tree/master/src/rules/declaration-block-no-ignored-properties

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  position: absolute;
  display: inline-block;
}

.bar {
  float: left;
  display: inline;
}
```

=> Warning should be shown as the display values are not applied in the browser.

> What stylelint configuration is needed to reproduce this issue?

---

> Which version of stylelint are you using?

---

> How are you running stylelint: CLI, PostCSS plugin, Node API?

---

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

---

> What did you expect to happen?

---

> What actually happened (e.g. what warnings or errors you are getting)?

---

**Comment by jeddy3** [comment]
@MattDiMu Thank you for the suggestion.

We've [previously](https://github.com/stylelint/stylelint/issues/1786) [discussed](https://github.com/stylelint/stylelint/pull/1906#issuecomment-247859344) the suitability of the `declaration-block-no-ignored-properties` rule. Your suggestion has further highlighted the complexity of this rule, and how it lacks a clear definition.

As such, I think your suggestion is likely to be the catalyst for removing this rule from core and into a community plugin. This is a good thing. It will promote experimentation, as well as more timely releases of versions with breaking changes. This could be a plugin you champion?

@stylelint/core I think this rule no longer adheres to [our criteria](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#let-everyone-know) and should be removed in `8.0.0`. Any objections?

**Comment by evilebottnawi** [comment]
@jeddy3 need add deprecation message to rule, i can move this rule in my account or orgs before release `8.0.0`

**Comment by evilebottnawi** [comment]
@jeddy3 or i can do this after add deprecation message :smile:

**Comment by davidtheclark** [comment]
Agreed. Let's move this rule into a plugin. @MattDiMu and @evilebottnawi would you like to collaborate on a single plugin that addresses it all?

**Comment by evilebottnawi** [comment]
@davidtheclark yep, i can move from `stylelint` after deprecation message (i think we should first add message then release - patch? and can move plugin out from stylelint)

**Comment by jeddy3** [comment]
Closing in favour of https://github.com/stylelint/stylelint/issues/2077 & https://github.com/stylelint/stylelint/issues/2078

**Comment by kristerkari** [comment]
@jeddy3 @evilebottnawi Did you ever create a plugin out of `declaration-block-no-ignored-properties`? 

I think that the rule is useful for keeping CSS tidy, so I'd be happy to help to create a plugin for it if there isn't one yet.

**Comment by evilebottnawi** [comment]
@kriansa no time for this now, but you can start with, if you need help feel free to ping me :+1:

**Comment by kristerkari** [comment]
Ok, I'll add a new plugin based on latest version of the rule before it got removed.

**Comment by kristerkari** [comment]
@evilebottnawi Done: https://github.com/kristerkari/stylelint-declaration-block-no-ignored-properties

**Comment by evilebottnawi** [comment]
@kristerkari can you invite me, in near future i do PR with other stuff, also i am creator this rule :smile:

**Comment by jeddy3** [comment]
Good stuff! Do you want to ping over a PR adding it [to the list](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/plugins.md)?

**Comment by kristerkari** [comment]
@jeddy3 Sure, PR done :)

**Comment by Mouvedia** [comment]
> @evilebottnawi Done: https://github.com/kristerkari/stylelint-declaration-block-no-ignored-properties

Are the replaced elements taken into account by the plugin?

**Comment by kristerkari** [comment]
@Mouvedia what do you mean by "replaced elements"?

**Comment by Mouvedia** [comment]
@kristerkari you can read about them here: #1786
